js/common/lib/tensor.ts

```diff
@@ -233,6 +233,29 @@ export interface TensorConstructor {
   new(data: Tensor.DataType, dims?: readonly number[]): Tensor;
   // #endregion
 }
+export interface TensorFactory {
+  /**
+   * Create a new tensor object from image object
+   * @param image - Image object holding the image data
+   */
+  fromImage(image: ImageData): Tensor;
+
+  /**
+   * Create a new tensor object from image object
+   * @param image - Image object holding the image data
+   */
+  fromImage(image: HTMLImageElement): Tensor;
+
+  /**
+   * Create a new tensor object from image object
+   * @param buffer - Extracted image buffer data - assuming RGBA format
+   * @param imgH - image height
+   * @param imgW - image width
+   * @param format - String - image buffer format
+   * @param norm - Boolean value for if to normalizing the buffer values between 1:-1 - default behavior is normalizing it by 255.
+   */
+   bufferToTensor(buffer: Uint8ClampedArray, imgH: number, imgW: number, format:string, norm:boolean): Tensor;
```

Comment:
- buffer: can support Uint8ClampedArray or Uint8Array
- imgH: please use more verbal name for parameter, like `imageHeight`
- need take a look at a few other API: is width usually put in front of height in parameter list in web API design? my feeling is 'yes' but I respect data so need to take a look at a few.
- format: try to include more details in the type instead of "string". eg. `format: 'rgba'|'rgb'|'rbg'` <--- this tells TypeScript that parameter format only has 3 valid value although its type is string.
- format and norm are all about how to pre-process the image before it goes to the model as input. make them and object `options` with properties inside.
- norm: is there going to be multiple ways to normalize an image? if so, we can change its type to string-based enum